### PR TITLE
Add new ldap attribute for Carbonio Chats meeting recording feature

### PR DESCRIPTION
**Pull request by nomanAli95**
A new LDAP attribute has been added to the account, cos, and domain level to enable or disable the meeting recording functionality on Carbonio Chats.

Support PR for **carbonio-ldap-utilities**:
https://github.com/Zextras/carbonio-ldap-utilities/pull/8

### CODE REVIEW

**Comment by keshavbhatt** [comment]
LGTM, we also require support PR for carbonio-ldap-utilities to complete this.

**Comment by nomanAli95** [comment]
> LGTM, we also require support PR for carbonio-ldap-utilities to complete this.

Yeah I know, I'm just waiting to merge this one to get the commit hash and use it to generate the correct timestamp for the required file name in carbonio-ldap-utilities

**Comment by keshavbhatt** [comment]
> I'm just waiting to merge this one to get the commit hash

No need to wait, just use the `git log -1 --pretty=format:%at store/conf/attrs/attrs.xml` in your mailbox branch and use that timestamp.

**Review by keshavbhatt — APPROVED** [review]

File: common/src/java/com/zimbra/common/account/ZAttrProvisioning.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 /**
  * AUTO-GENERATED. DO NOT EDIT.
- * 
+ *
```
[the remark this answers is not in the document]
Comment (frisonisland) [reply common/src/java/com/zimbra/common/account/ZAttrProvisioning.java:14]: Ok, it's not just this file, also all other ZAttr* files are modified by generate-getters.
Anyway I agree, let's check on it next time :+1:

File: store/src/java/com/zimbra/cs/account/ZAttrCos.java [hunk 2]
```diff
@@ -47185,6 +47185,68 @@ public Map<String,Object> unsetPasswordBlockCommonEnabled(Map<String,Object> att
...
+  /**
+   * Enable video server recording for Carbonio chats
+   *
+   * @param carbonioChatsVideoServerRecordingEnabled new value
+   * @param attrs existing map to populate, or null to create a new map
+   * @return populated map to pass into Provisioning.modifyAttrs
+   */
+  @ZAttr(id=3091)
+  public Map<String,Object> setCarbonioChatsVideoServerRecordingEnabled(boolean carbonioChatsVideoServerRecordingEnabled, Map<String,Object> attrs) {
+    if (attrs == null) attrs = new HashMap<String,Object>();
```
Comment (Polpetta) [inline store/src/java/com/zimbra/cs/account/ZAttrCos.java:47220]: If without parenthesis

File: store/src/java/com/zimbra/cs/account/ZAttrCos.java [hunk 3]
```diff
@@ -47185,6 +47185,68 @@ public Map<String,Object> unsetPasswordBlockCommonEnabled(Map<String,Object> att
...
+
+  /**
+   * Enable video server recording for Carbonio chats
+   *
+   * @param attrs existing map to populate, or null to create a new map
+   * @return populated map to pass into Provisioning.modifyAttrs
+   */
+  @ZAttr(id=3091)
+  public Map<String,Object> unsetCarbonioChatsVideoServerRecordingEnabled(Map<String,Object> attrs) {
+    if (attrs == null) attrs = new HashMap<String,Object>();
```
Comment (Polpetta) [inline store/src/java/com/zimbra/cs/account/ZAttrCos.java:47245]: If without parenthesis

File: store/src/java/com/zimbra/cs/account/ZAttrAccount.java [hunk 4]
```diff
@@ -61785,6 +61785,68 @@ public Map<String,Object> unsetPasswordBlockCommonEnabled(Map<String,Object> att
...
+
+    /**
+     * Enable video server recording for Carbonio chats
+     *
+     * @param attrs existing map to populate, or null to create a new map
+     * @return populated map to pass into Provisioning.modifyAttrs
+     */
+    @ZAttr(id=3091)
+    public Map<String,Object> unsetCarbonioChatsVideoServerRecordingEnabled(Map<String,Object> attrs) {
+      if (attrs == null) attrs = new HashMap<String,Object>();
```
Comment (Polpetta) [inline store/src/java/com/zimbra/cs/account/ZAttrAccount.java:61845]: if without parenthesis

File: store/src/java/com/zimbra/cs/account/ZAttrDomain.java [hunk 5]
```diff
@@ -24626,6 +24626,68 @@ public Map<String,Object> unsetZimletUserPropertiesMaxNumEntries(Map<String,Obje
...
+    /**
+     * Enable video server recording for Carbonio chats
+     *
+     * @param carbonioChatsVideoServerRecordingEnabled new value
+     * @param attrs existing map to populate, or null to create a new map
+     * @return populated map to pass into Provisioning.modifyAttrs
+     */
+    @ZAttr(id=3091)
+    public Map<String,Object> setCarbonioChatsVideoServerRecordingEnabled(boolean carbonioChatsVideoServerRecordingEnabled, Map<String,Object> attrs) {
+        if (attrs == null) attrs = new HashMap<String,Object>();
```
Comment (Polpetta) [inline store/src/java/com/zimbra/cs/account/ZAttrDomain.java:24661]: Same here

**Review by Polpetta — CHANGES_REQUESTED** [review]
I don't know if those files are effectively auto-generated, but if not there are a bunch of missing parenthesis!

**Comment by keshavbhatt** [comment]
> I don't know if those files are effectively auto-generated, but if not there are a bunch of missing parenthesis!

just verified with auto generator, it also likes no parentheses;

**Comment by nomanAli95** [comment]
> > I don't know if those files are effectively auto-generated, but if not there are a bunch of missing parenthesis!
> 
> just verified with auto generator, it also likes no parentheses;

If you want I can add them, no problem

**Comment by keshavbhatt** [comment]
> > > I don't know if those files are effectively auto-generated, but if not there are a bunch of missing parenthesis!
> > 
> > 
> > just verified with auto generator, it also likes no parentheses;
> 
> If you want I can add them, no problem

Will leave it up to you. I am okay with this exceptional case(We prefer parenthesis). We will be using the Auto generator in the future so no issues.

**Review by frisonisland — APPROVED** [review]

**Comment by Polpetta** [comment]
I guess...we could merge them anyway :sweat_smile:

**Pull request merged**